File: javascript/rna-transcription/rna-transcription.js
export function toRna (dna) {
  if (dna.length === 0) return ''

  // test that dna contains only GCTA characters
  const re = /^[GCTA]+$/
  if (!re.test(dna)) {
    throw new Error('Invalid input DNA.')
  }

  const translate = {
    'G': 'C',
    'C': 'G',
    'T': 'A',
    'A': 'U'
  }

  let dnaArr = dna.split('')
  let rna = dnaArr.map(c => translate[c])
  return rna.join('')
}

File: python/rna-transcription/rna_transcription.py
'''
https://docs.python.org/3.6/library/stdtypes.html?highlight=translate#str.translate
https://docs.python.org/3.6/library/stdtypes.html?highlight=translate#str.maketrans
'''
import re


def to_rna(dna_strand):
    if re.search('^[GCTA]+$', dna_strand):
        return dna_strand.translate(str.maketrans('GCTA', 'CGAU'))
    else:
        raise ValueError("DNA strand invalid, empty or incorrect nucleotide")
